src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -1714,13 +1714,20 @@ private static async Task<IList<CompatibilityCheckResult>> VerifyCompatibilityAs
 
             Tuple<bool, List<RestoreTargetGraph>, RuntimeGraph> result = null;
             bool failed = false;
+
+            Dictionary<NuGetFramework, string> targetFrameworkToAlias = projectFrameworkRuntimePairs.Where(e => string.IsNullOrEmpty(e.RuntimeIdentifier)).ToDictionary(
```

Comment:
perf nitpick: How sensitive are we to minor perf possibilities in restore?  LINQ methods like `.Where` and `.ToDictionary` will allocate enumerators, and all the delegates will probably allocate as well (at least they historically would have. I think it's theoretically possible for the compiler to determine when a static delegate could be pre-compiled).

This code could avoid these 2 to 5 allocations by writing traditional `for` loops (sometimes `foreach` as well, if the collection has a struct enumerator). Pretty much anywhere other than restore, it wouldn't matter. For restore, I don't know if we always want to be proactive about minimizing easy allocations.